libvmi/driver/xen/xen_events_legacy.c

```diff
@@ -119,7 +119,7 @@ static inline int get_mem_event_42(xen_mem_event_t *mem_event, mem_event_42_requ
     req_cons = back_ring->req_cons;
 
     // Copy request
-    memcpy(req, RING_GET_REQUEST(back_ring, req_cons), sizeof(*req));
+    memcpy(req, RING_GET_REQUEST(back_ring, req_cons), sizeof(mem_event_42_request_t));
```

Comment:
Yay coverity. Excellent, thanks

On Wed, Aug 24, 2016 at 11:16 AM, Tamas K Lengyel <notifications@github.com>
wrote:

> In libvmi/driver/xen/xen_events_legacy.c
> <https://github.com/libvmi/libvmi/pull/365#discussion_r76076828>:
>
> > @@ -119,7 +119,7 @@ static inline int get_mem_event_42(xen_mem_event_t *mem_event, mem_event_42_requ
> >      req_cons = back_ring->req_cons;
> >
> >      // Copy request
> > -    memcpy(req, RING_GET_REQUEST(back_ring, req_cons), sizeof(*req));
> > +    memcpy(req, RING_GET_REQUEST(back_ring, req_cons), sizeof(mem_event_42_request_t));
>
> On a closer look it was actually a problem with the ring definition in our
> internal ABI. Fixed.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/libvmi/libvmi/pull/365/files/db8ce41b48bd777fae7fac509b26c5dd028a58d5#r76076828>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA5auFLioQY4B-xGfpWZkWVpV81USdoPks5qjGA_gaJpZM4Jo6z0>
> .
>